src/arraymancer/tensor/aggregate.nim

```diff
@@ -337,6 +342,116 @@ proc cumprod*[T](arg: Tensor[T], axis: int = 0): Tensor[T] = # from hugogranstro
     else:
       temp[_] = result.atAxisIndex(axis, i-1) *. tAxis
 
+proc diff*[T](arg: Tensor[T], n=1, axis: int = -1): Tensor[T] =
```

Comment:
I usually like to reuse Numpy names but it's not very expressive, I think `discrete_diff` would be more expressive.

Reply:
I don't find "diff" a confusing name and it is used not just by numpy but also by Matlab and Julia. I can change it if you want but IMHO it is pretty clear.

Reply:
I think `diff_discrete` so that those looking for `diff` get the autocomplete and can see "ah this is it"

Reply:
Change made and PR updated. Also improve the documentation of the function a bit.